File: module-common/src/main/java/com/geekhome/common/configurable/LongFieldBuilder.java
package com.geekhome.common.configurable;

public class LongFieldBuilder implements FieldBuilder<Long> {

    @Override
    public Long fromPersistableString(String value) {
        if (value == null) {
            return null;
        }

        try {
            return Long.parseLong(value);
        } catch (NumberFormatException nfe) {
            return null;
        }
    }

    @Override
    public String toPersistableString(Long value) {
        return value.toString();
    }
}

File: modules/module-domain/src/main/java/eu/geekhome/domain/configurable/LongFieldBuilder.kt
package eu.geekhome.domain.configurable

import java.lang.NumberFormatException

class LongFieldBuilder : FieldBuilder<Long?> {
    override fun fromPersistableString(value: String?): Long {
        return if (value == null) {
            0L
        } else try {
            value.toLong()
        } catch (nfe: NumberFormatException) {
            0L
        }
    }

    override fun toPersistableString(value: Long?): String {
        return value.toString()
    }
}